Tasks/FtpUploadV2/ftpuploadtask.ts

```diff
@@ -0,0 +1,341 @@
+import * as tl from "azure-pipelines-task-lib/task";
+import * as ftp from "basic-ftp";
+import * as fs from "fs";
+import * as path from "path";
+import * as url from "url";
+
+class FtpOptions {
+    // url
+    serverEndpointUrl: url.Url;
+
+    // credentials
+    username: string;
+    password: string;
+
+    // other standard options
+    rootFolder: string;
+    filePatterns: string[];
+    remotePath: string;
+
+    // advanced options
+    clean: boolean;
+    cleanContents: boolean;
+    preservePaths: boolean;
+    trustSSL: boolean;
+}
+
+class ProgressTracking {
+    ftpOptions: FtpOptions = null;
```

Comment:
Looks like these get initialized in the constructor?

Just use 

`constructor(private ftpOptions: FtpOptions, private fileCount: number)`